storage/secondary.go

```diff
@@ -47,18 +48,18 @@ func newSecondaryQuerierFromChunk(cq ChunkQuerier) genericQuerier {
 	return &secondaryQuerier{genericQuerier: newGenericQuerierFromChunk(cq)}
 }
 
-func (s *secondaryQuerier) LabelValues(name string, matchers ...*labels.Matcher) ([]string, Warnings, error) {
+func (s *secondaryQuerier) LabelValues(name string, matchers ...*labels.Matcher) ([]string, annotations.Annotations, error) {
 	vals, w, err := s.genericQuerier.LabelValues(name, matchers...)
 	if err != nil {
-		return nil, append([]error{err}, w...), nil
+		return nil, annotations.InitAnnotationsAndMerge(err, w), nil
```

Comment:
I don't think you have to create a new `Annotations` here. Wouldn't the following work?
```Go
return nil, w.Add(err), nil
```